datanode/gateway/graphql/resolvers.go

```diff
@@ -1291,28 +1291,36 @@ func (r *myQueryResolver) Statistics(ctx context.Context) (*vegaprotoapi.Statist
 	return resp.GetStatistics(), nil
 }
 
-func (r *myQueryResolver) HistoricBalances(ctx context.Context, filter *v2.AccountFilter, groupBy []*v2.AccountField, dateRange *v2.DateRange, pagination *v2.Pagination) (*v2.AggregatedBalanceConnection, error) {
-	gb := make([]v2.AccountField, len(groupBy))
-	for i, g := range groupBy {
-		if g == nil {
-			return nil, errors.New("nil group by")
-		}
-		gb[i] = *g
+func (r *myQueryResolver) BalanceHistory(
+	ctx context.Context,
+	filter *v2.AccountFilter,
+	sumParties, sumMarkets, sumTypes *bool,
+	dateRange *v2.DateRange,
+	pagination *v2.Pagination,
+) (*v2.AggregatedBalanceConnection, error) {
+	req := &v2.ListBalanceHistoryRequest{
+		Filter:           filter,
+		SumAcrossParties: unboxBool(sumParties),
+		SumAcrossMarkets: unboxBool(sumMarkets),
+		SumAcrossTypes:   unboxBool(sumTypes),
+		DateRange:        dateRange,
+		Pagination:       pagination,
 	}
 
-	req := &v2.GetBalanceHistoryRequest{}
-	req.GroupBy = gb
-	req.Filter = filter
-	req.DateRange = dateRange
-	req.Pagination = pagination
-
-	resp, err := r.tradingDataClientV2.GetBalanceHistory(ctx, req)
+	resp, err := r.tradingDataClientV2.ListBalanceHistory(ctx, req)
 	if err != nil {
 		return nil, err
 	}
 	return resp.GetBalances(), nil
 }
 
+func unboxBool(b *bool) bool {
+	if b == nil {
+		return false
+	}
+	return *b
+}
```

Comment:
I guess we could add this as some function that we could reuse eventually in the ptr package?
Something like so I suppose:
```
func UnBox[T any](v *T) (out T) {
	if v != nil {
		return *v
	}
	return
}
```

Reply:
Yeah that's quite nice, I'll do that.. need to fix some linty stuff anyway

Reply:
Added this to the 'ptr' package